### ISSUE TITLE
Names get mangled if multiple tables are exported at the same time

### ISSUE THREAD

**Issue by qwesda** [jakob/Postico#838]
This is a fun one ...

Somehow the names of exported tables get mangled. Strangely it only happens if there is a significant amount of data.

Here is how I could reproduce the bug:

1. create some dummy data

```
CREATE TEMP TABLE foo AS SELECT md5(clock_timestamp()::text) AS foo FROM generate_series(1, 1000000) AS _(x);
CREATE TEMP TABLE bar AS SELECT md5(clock_timestamp()::text) AS bar FROM generate_series(1, 1000000) AS _(x);
CREATE TEMP TABLE baz AS SELECT md5(clock_timestamp()::text) AS baz FROM generate_series(1, 1000000) AS _(x);
```

2. export the 3 tables at the same time
<img width="356" alt="image" src="https://user-images.githubusercontent.com/384993/231853592-73d45733-7544-4143-8e4c-a81b79fe9f3e.png">

3. inspect exported files

```
> head -n 1 *
==> bar 2.csv <==
"baz"

==> bar.csv <==
"bar"

==> baz.csv <==
"foo"
```

Postico version:  
Version 2.0.2 dev (9610)

**Comment by jakob** [comment]
I wasn't able to reproduce it, but after reviewing the code I think I found the bug. There's a race condition in the progress reporting code that can reset the list of table names, so Postico starts using table names again that it has already used.

I've addressed the issue in build 9615. I also found a bug in the progress indicator. The new build should have a proper status bar again, instead of the indeterminate bar going back and forth.
